hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java

```diff
@@ -75,43 +85,59 @@ public String getTriggerId() {
 
     public void setTriggerId(String triggerId) {
         this.triggerId = triggerId;
+        updateId();
     }
 
-    public String getConditionId() {
-        conditionId = triggerId + "-" + conditionSetSize + "-" + conditionSetIndex;
-        return conditionId;
+    public Mode getTriggerMode() {
+        return triggerMode;
     }
 
-    public void setConditionId(String conditionId) {
-        this.conditionId = conditionId;
+    public void setTriggerMode(Mode triggerMode) {
+        this.triggerMode = triggerMode;
+        updateId();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        Condition condition = (Condition) o;
-
-        if (conditionSetIndex != condition.conditionSetIndex) return false;
-        if (conditionSetSize != condition.conditionSetSize) return false;
-        if (triggerId != null ? !triggerId.equals(condition.triggerId) : condition.triggerId != null) return false;
+    public String getConditionId() {
+        return conditionId;
+    }
 
-        return true;
+    private void updateId() {
+        StringBuilder sb = new StringBuilder(triggerId);
+        sb.append("-").append(triggerMode.ordinal());
```

Comment:
I think that instead to save an ordinal (that is hard to remind) it can be also good to store the literal of the triggerMode, I guess that at this time it's ok and we can gain for clarity.